packages/go_router/lib/src/redirection.dart

```diff
@@ -71,14 +71,14 @@ RouteMatchList redirect(RouteMatchList prevMatchList,
     if (top == null) {
       break;
     }
-    final String? topRouteLocation = top.route.redirect(
+    final String? topRouteLocation = (top.route as GoRoute).redirect(
```

Comment:
since there are multiple recast, you may as well to recast to a variable to avoid all the recast, I remember recast is rather expansive in web

Reply:
e304529e5c3e4572f345b053877fa42c3abbb2dc